INT64_MIN/MAX not defined for newer g++

### Description

While cross compiling for arm w/ newer versions of g++ INT64_MAX, INT64_MIN, UINT64, INT32_MIN, INT32_MAX, UINT32_MAX aren't defined.  I found the following:
/* The ISO C99 standard specifies that in C++ implementations these
   macros should only be defined if explicitly requested.  */
#if !defined __cplusplus || defined __STDC_LIMIT_MACROS

Rather than adding the #define to make this work I created a patch that uses the std::numeric_limits<int64_t>::max() 
I guess a "compatibility" include file could be created to work around this issue; but it wasn't obvious to me how to do this.

After applying the patch I ran all the tests and they all passed.
[int64_min_max.patch.txt](https://github.com/nlohmann/json/files/9483669/int64_min_max.patch.txt)


### Reproduction steps

git clone https://github.com/nlohmann/json.git
cd json
git checkout v3.11.2
mkdir build
cd build
cmake -DCMAKE_TOOLCHAIN_FILE=../../aarch64.cmake -DCMAKE_INSTALL_PREFIX=/opt/nxpimx8/usr/local ..
make


### Expected vs. actual results

See compiler errors below.

### Minimal code example

_No response_

### Error messages

```Shell
json/tests/src/unit-to_chars.cpp:466:23: error: 'INT64_MIN' was not declared in this scope
  466 |         check_integer(INT64_MIN, "-9223372036854775808");
      |                       ^~~~~~~~~
/home/priestwilliaml/newbuild/build/json/tests/src/unit-to_chars.cpp:467:23: error: 'INT64_MAX' was not declared in this scope
  467 |         check_integer(INT64_MAX, "9223372036854775807");
      |                       ^~~~~~~~~

json/tests/src/unit-cbor.cpp: In function 'void _DOCTEST_ANON_FUNC_7()':
/home/priestwilliaml/newbuild/build/json/tests/src/unit-cbor.cpp:176:39: error: 'INT64_MIN' was not declared in this scope
  176 |                     numbers.push_back(INT64_MIN);
      |                                       ^~~~~~~~~

json/tests/src/unit-msgpack.cpp:511:39: error: 'INT64_MIN' was not declared in this scope
  511 |                     numbers.push_back(INT64_MIN);
      |                                       ^~~~~~~~~

json/tests/src/unit-regression1.cpp: In function 'void _DOCTEST_ANON_FUNC_7()':
/home/priestwilliaml/newbuild/build/json/tests/src/unit-regression1.cpp:880:51: error: 'INT64_MIN' was not declared in this scope
  880 |         CHECK(j1.get<json::number_integer_t>() == INT64_MIN);
      |                                                   ^~~~~~~~~
```


### Compiler and operating system

linux g++ 11.2.0 crosscompiling for ARM 64

### Library version

v3.11.2 from github

### Validation

- [X] The bug also occurs if the latest version from the [`develop`](https://github.com/nlohmann/json/tree/develop) branch is used.
- [X] I can successfully [compile and run the unit tests](https://github.com/nlohmann/json#execute-unit-tests).

--- a/tests/src/unit-bson.cpp
+++ b/tests/src/unit-bson.cpp
@@ -12,6 +12,7 @@
 using nlohmann::json;
 
 #include <fstream>
+#include <limits>
 #include <sstream>
 #include "make_test_data_available.hpp"
 #include "test_utils.hpp"
@@ -865,7 +866,7 @@ TEST_CASE("BSON numerical data")
             {
                 std::vector<int64_t> const numbers
                 {
-                    INT64_MIN,
+                    (std::numeric_limits<int64_t>::min)(),
                     -1000000000000000000LL,
                     -100000000000000000LL,
                     -10000000000000000LL,
@@ -875,7 +876,7 @@ TEST_CASE("BSON numerical data")
                     -1000000000000LL,
                     -100000000000LL,
                     -10000000000LL,
-                    static_cast<std::int64_t>(INT32_MIN) - 1,
+                    static_cast<std::int64_t>((std::numeric_limits<std::int32_t>::min)()) - 1,
                 };
 
                 for (const auto i : numbers)
@@ -923,7 +924,7 @@ TEST_CASE("BSON numerical data")
             {
                 std::vector<int32_t> const numbers
                 {
-                    INT32_MIN,
+                    (std::numeric_limits<int32_t>::min)(),
                     -2147483647L,
                     -1000000000L,
                     -100000000L,
@@ -947,7 +948,7 @@ TEST_CASE("BSON numerical data")
                     100000000L,
                     1000000000L,
                     2147483646L,
-                    INT32_MAX
+                    (std::numeric_limits<int32_t>::max)()
                 };
 
                 for (const auto i : numbers)
@@ -990,7 +991,7 @@ TEST_CASE("BSON numerical data")
             {
                 std::vector<int64_t> const numbers
                 {
-                    INT64_MAX,
+                    (std::numeric_limits<int64_t>::max)(),
                     1000000000000000000LL,
                     100000000000000000LL,
                     10000000000000000LL,
@@ -1000,7 +1001,7 @@ TEST_CASE("BSON numerical data")
                     1000000000000LL,
                     100000000000LL,
                     10000000000LL,
-                    static_cast<std::int64_t>(INT32_MAX) + 1,
+                    static_cast<std::int64_t>((std::numeric_limits<int32_t>::max)()) + 1,
                 };
 
                 for (const auto i : numbers)
@@ -1062,7 +1063,7 @@ TEST_CASE("BSON numerical data")
                     100000000ULL,
                     1000000000ULL,
                     2147483646ULL,
-                    static_cast<std::uint64_t>(INT32_MAX)
+                    static_cast<std::uint64_t>((std::numeric_limits<int32_t>::max)())
                 };
 
                 for (const auto i : numbers)
@@ -1105,9 +1106,9 @@ TEST_CASE("BSON numerical data")
             {
                 std::vector<std::uint64_t> const numbers
                 {
-                    static_cast<std::uint64_t>(INT32_MAX) + 1,
+                    static_cast<std::uint64_t>((std::numeric_limits<std::int32_t>::max)()) + 1,
                     4000000000ULL,
-                    static_cast<std::uint64_t>(UINT32_MAX),
+                    static_cast<std::uint64_t>((std::numeric_limits<std::uint32_t>::max)()),
                     10000000000ULL,
                     100000000000ULL,
                     1000000000000ULL,
@@ -1117,7 +1118,7 @@ TEST_CASE("BSON numerical data")
                     10000000000000000ULL,
                     100000000000000000ULL,
                     1000000000000000000ULL,
-                    static_cast<std::uint64_t>(INT64_MAX),
+                    static_cast<std::uint64_t>((std::numeric_limits<std::int64_t>::max)()),
                 };
 
                 for (const auto i : numbers)
@@ -1163,11 +1164,11 @@ TEST_CASE("BSON numerical data")
             {
                 std::vector<std::uint64_t> const numbers
                 {
-                    static_cast<std::uint64_t>(INT64_MAX) + 1ULL,
+                    static_cast<std::uint64_t>((std::numeric_limits<std::int64_t>::max)()) + 1ULL,
                     10000000000000000000ULL,
                     18000000000000000000ULL,
-                    UINT64_MAX - 1ULL,
-                    UINT64_MAX,
+                    (std::numeric_limits<std::uint64_t>::max)() - 1ULL,
+                    (std::numeric_limits<std::uint64_t>::max)(),
                 };
 
                 for (const auto i : numbers)

--- a/tests/src/unit-cbor.cpp
+++ b/tests/src/unit-cbor.cpp
@@ -15,6 +15,7 @@ using nlohmann::json;
 #include <sstream>
 #include <iomanip>
 #include <iostream>
+#include <limits>
 #include <set>
 #include "make_test_data_available.hpp"
 #include "test_utils.hpp"
@@ -174,7 +175,7 @@ TEST_CASE("CBOR")
                 {
                     const std::vector<int64_t> numbers
                     {
-                        INT64_MIN,
+                        (std::numeric_limits<int64_t>::min)(),
                         -1000000000000000000,
                         -100000000000000000,
                         -10000000000000000,

--- a/tests/src/unit-msgpack.cpp
+++ b/tests/src/unit-msgpack.cpp
@@ -17,6 +17,7 @@ using nlohmann::json;
 #include <fstream>
 #include <sstream>
 #include <iomanip>
+#include <limits>
 #include <set>
 #include "make_test_data_available.hpp"
 #include "test_utils.hpp"
@@ -526,7 +527,7 @@ TEST_CASE("MessagePack")
                 {
                     std::vector<int64_t> const numbers
                     {
-                        INT64_MIN,
+                        (std::numeric_limits<int64_t>::min)(),
                         -2147483649LL,
                     };
                     for (auto i : numbers)

--- a/tests/src/unit-regression1.cpp
+++ b/tests/src/unit-regression1.cpp
@@ -21,6 +21,7 @@ using nlohmann::json;
 #include <fstream>
 #include <sstream>
 #include <list>
+#include <limits>
 #include <cstdio>
 #include "make_test_data_available.hpp"
 
@@ -877,12 +878,12 @@ TEST_CASE("regression tests 1")
         // original test case
         json const j1 = json::parse("-9223372036854775808");
         CHECK(j1.is_number_integer());
-        CHECK(j1.get<json::number_integer_t>() == INT64_MIN);
+        CHECK(j1.get<json::number_integer_t>() == (std::numeric_limits<std::int64_t>::min)());
 
         // edge case (+1; still an integer)
         json const j2 = json::parse("-9223372036854775807");
         CHECK(j2.is_number_integer());
-        CHECK(j2.get<json::number_integer_t>() == INT64_MIN + 1);
+        CHECK(j2.get<json::number_integer_t>() == (std::numeric_limits<std::int64_t>::min)() + 1);
 
         // edge case (-1; overflow -> floats)
         json const j3 = json::parse("-9223372036854775809");

--- a/tests/src/unit-windows_h.cpp
+++ b/tests/src/unit-windows_h.cpp
@@ -7,6 +7,8 @@
 // SPDX-License-Identifier: MIT
 
 #include "doctest_compatibility.h"
+#undef WIN32_LEAN_AND_MEAN
+#undef NOMINMAX
 
 #ifdef _WIN32
     #include <windows.h>

